src/cmakeProject.ts

```diff
@@ -266,11 +266,24 @@ export class CMakeProject {
             return undefined;
         }
         log.debug(localize('resolving.config.preset', 'Resolving the selected configure preset'));
+
+        // We want to use the original unexpanded preset file to apply the dev env in expandConfigurePreset
+        // we have to first check if the preset is valid in expandedPresetsFiles since we won't be expanding the whole file here, only the path up for this preset
+        if (!preset.getPresetByName(preset.configurePresets(this.folderPath), configurePreset)) {
+            return undefined;
+        }
+
+        // TODO: move applyDevEnv here to decouple from expandConfigurePreset
```

Comment:
The TODO is fine, besides our offline discussion about how in general we should try to avoid too many of these so we don't just continually grow TODO's. We discussed why this case is fine 👍, just figured I'd comment on it again.